dashboard/app/models/course.rb

```diff
@@ -18,6 +18,32 @@ class Course < ApplicationRecord
   has_one :plc_course, class_name: 'Plc::Course'
   has_many :course_scripts, -> {order('position ASC')}
 
+  validates :name,
+    presence: true,
+    uniqueness: {case_sensitive: false}
+
+  class CourseNameValidator < ActiveModel::Validator
+    def validate(course)
+      return if course.plc_course
+
+      unless /\A[a-z0-9\-]+\z/ =~ course.name
+        course.errors[:base] << 'can only contain lowercase letters, numbers and dashes'
+      end
+    end
+  end
+
+  validates_with CourseNameValidator
```

Comment:
Why this format instead of `validates :name` (with `unless: :plc_course`)?  It looks like there's [common functionality in `Script`](https://github.com/code-dot-org/code-dot-org/blob/88a149b3362850761d14424655c2a585d4e82f92/dashboard/app/models/script.rb#L42-L49) we could maybe combine into a single concern.  If we want to start standardizing on this format for all names that appear in URLs...